### ISSUE TITLE
First-time deployment fails with "the server is currently unable to handle the request"

### ISSUE THREAD

**Issue by drigz** [googlecloudrobotics/core#17]
Sometimes a first-time deployment fails with:

```
apply files: set default namespaces: discover server resources: unable to retrieve the complete list of server APIs: metrics.k8s.io/v1beta1: the server is currently unable to handle the request
```

Note the `metrics.k8s.io/v1beta1`; this is not the same as another soon-to-be-fixed issue that complains about `admission.certmanager.k8s.io`.

This appears to be because a certain webhook is not yet available in a newly created cluster, so when synk enumerates all APIs it hits this error and treats it as fatal.

### Workaround

Re-run deploy.sh.

### Idea for a fix

Change [populateNamespaces](https://github.com/googlecloudrobotics/core/blob/59b6929e27b00380fdf2d1a7f0151b874baf618e/src/go/pkg/synk/synk.go#L415) to only report errors for API groups that the resources reference.

**Comment by ensonic** [comment]
The proposed fix will definitely help. Thanks for looking into it.

**Comment by drigz** [comment]
Fixed by 38e9c218992f5af499436face9704f7a9d089c03 (add retries to synk command).
